File: CountryExplorerBE/CountryExplorer/Adapters/Models/Country.cs
namespace CountryExplorer.Adapters.Models
{
    public class Country
    {
        public Name Name { get; set; }

        public Dictionary<string, Currency> Currencies { get; set; }

        public IReadOnlyList<string> Capital { get; set; }

        public string Region { get; set; }

        public Dictionary<string, string> Languages { get; set; }

        public Dictionary<string, string> Flags { get; set; }

        public Dictionary<string, string> Maps { get; set; }
    }
}

File: CountryExplorerApp/src/models/country.ts
export class Country {
    name?: Name;
    currency?: Record<string, Currency> = {};
    capital?: string[];
    region?: string;
    image?: string;
    mapLink?: string;
    language?: Record<string, string> = {};
}

export class Name {
    common?: string;
    official?: string;
}

export class Currency {
    name?: string;
    symbol?: string;
}